validator/keymanager/v2/direct/direct.go

```diff
@@ -467,7 +490,90 @@ func (dr *Keymanager) createAccountsKeystore(
 	}, nil
 }
 
-func (dr *Keymanager) askUntilPasswordConfirms(
+func (dr *Keymanager) listenForFileChanges(ctx context.Context, filePath string) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.WithError(err).Error("Could not initialize file watcher")
+		return
+	}
+	defer func() {
+		if err := watcher.Close(); err != nil {
+			log.WithError(err).Error("Could not close file watcher")
+			return
+		}
+	}()
+	if err := watcher.Add(filePath); err != nil {
+		log.WithError(err).Errorf("Could not add file %s to file watcher", filePath)
+		return
+	}
+	for {
+		select {
+		case event := <-watcher.Events:
+			// If a file was modified, we attempt to read that file
+			// and parse it into our accounts store.
+			if event.Op&fsnotify.Write == fsnotify.Write {
+				fileBytes, err := ioutil.ReadFile(event.Name)
+				if err != nil {
+					log.WithError(err).Errorf("Could not read file at path: %s", event.Name)
+					continue
+				}
+				accountsKeystore := &v2keymanager.Keystore{}
+				if err := json.Unmarshal(fileBytes, accountsKeystore); err != nil {
+					log.WithError(
+						err,
+					).Errorf("Could not read valid, EIP-2335 keystore json file at path: %s", event.Name)
+					continue
+				}
+				if err := dr.reloadAccountsFromKeystore(accountsKeystore); err != nil {
+					log.WithError(
+						err,
+					).Error("Could not replace the accounts store from keystore file")
+				}
+			}
+		case err := <-watcher.Errors:
+			log.WithError(err).Errorf("Could not watch for file changes for: %s", filePath)
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+// Replaces the accounts store struct in the direct keymanager with
+// the contents of a keystore file by decrypting it with the accounts password.
+func (dr *Keymanager) reloadAccountsFromKeystore(keystore *v2keymanager.Keystore) error {
+	// If no one is subscribing to account changes, no need to proceed.
+	if dr.accountsChangedFeed == nil {
+		return nil
+	}
+	decryptor := keystorev4.New()
+	encodedAccounts, err := decryptor.Decrypt(keystore.Crypto, dr.accountsPassword)
+	if err != nil {
+		return errors.Wrapf(err, "could not decrypt keystore file with public key %s", keystore.Pubkey)
+	}
+	newAccountsStore := &AccountStore{}
+	if err := json.Unmarshal(encodedAccounts, newAccountsStore); err != nil {
+		return err
+	}
+	dr.lock.Lock()
+	dr.keysCache = make(map[[48]byte]bls.SecretKey)
+	dr.accountsStore = newAccountsStore
+	pubKeys := make([][48]byte, len(dr.accountsStore.PublicKeys))
+	for i := 0; i < len(dr.accountsStore.PrivateKeys); i++ {
+		privKey, err := bls.SecretKeyFromBytes(dr.accountsStore.PrivateKeys[i])
+		if err != nil {
+			return errors.Wrap(err, "could not initialize private key")
+		}
+		pubKeyBytes := privKey.PublicKey().Marshal()
+		pubKeys[i] = bytesutil.ToBytes48(pubKeyBytes)
+		dr.keysCache[bytesutil.ToBytes48(pubKeyBytes)] = privKey
+	}
+	dr.lock.Unlock()
```

Comment:
Should probably use a defer unlock with a flag that you can clear in normal cases when you unlock it before calling Send(), just in case an exception happens while populating the keysCache.